backend/internal/application/service.go

```diff
@@ -73,56 +77,47 @@ func newApplicationService(appStore applicationStoreInterface,
 func (as *applicationService) CreateApplication(app *model.ApplicationDTO) (*model.ApplicationDTO,
 	*serviceerror.ServiceError) {
 	logger := log.GetLogger().With(log.String(log.LoggerKeyComponentName, "ApplicationService"))
-
-	if app == nil {
-		return nil, &ErrorApplicationNil
-	}
-	if app.Name == "" {
-		return nil, &ErrorInvalidApplicationName
-	}
-
-	// Check if an application with the same name already exists
-	existingApp, appCheckErr := as.appStore.GetApplicationByName(app.Name)
-	if appCheckErr != nil && !errors.Is(appCheckErr, model.ApplicationNotFoundError) {
-		logger.Error("Failed to check existing application by name", log.Error(appCheckErr),
-			log.String("appName", app.Name))
-		return nil, &ErrorInternalServerError
-	}
-	if existingApp != nil {
-		return nil, &ErrorApplicationAlreadyExistsWithName
+	if config.GetThunderRuntime().Config.ImmutableResources.Enabled {
+		return nil, &filebasedruntime.ErrorImmutableResourceCreateOperation
```

Comment:
shouldn't this added to other service level write functions as well

Reply:
Error will be returned from the store level, but added service level check as well.